File: 771.宝石与石头/771.宝石与石头.py
'''
Descroption: LeetCode 771. 宝石与石头
Author: EmoryHuang
Date: 2021-08-25 15:43:55
Method:
简单题，哈希表计数
'''


class Solution:
    # def numJewelsInStones(self, jewels: str, stones: str) -> int:
    #     ans = 0
    #     c = Counter(stones)
    #     for key, val in c.items():
    #         if key in jewels:
    #             ans += val
    #     return ans

    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        return sum(s in jewels for s in stones)

File: Solution/771.宝石与石头/771.宝石与石头.java
/*
 * @Descroption: LeetCode 771. 宝石与石头
 * @Author: EmoryHuang
 * @Date: 2023-07-24 09:24:04
 * @Method:
 * 哈希表
 */

import java.util.*;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> hash = new HashSet<>();
        for (char ch : jewels.toCharArray()) {
            hash.add(ch);
        }
        int ans = 0;
        for (char ch : stones.toCharArray()) {
            if (hash.contains(ch)) {
                ans++;
            }
        }
        return ans;
    }
}